### PR TITLE
Feature/monthly leader board

### PR DESCRIPTION
**Pull request by emonnin-epfl**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java [hunk 1]
```diff
@@ -20,23 +21,28 @@
...
-    private final ImageView[] images = new ImageView[MAX_RANK];
+    public final Database db = new Database();
+    public final FirebaseAuth mAuth = FirebaseAuth.getInstance();
+    public int MIN_RANK;
+    public Button[] tab;
+    public PopupMenu.OnMenuItemClickListener[] menus;
+    public String[] ids;
+    public PopupMenu.OnMenuItemClickListener[] myMenus;
+    public ImageView[] images;
+    public final static int STRING_MAX_LEN = 40;
```
Comment (Aco-Hub) [inline app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:45]: unused constant

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java [hunk 2]
```diff
@@ -154,29 +172,29 @@ private PopupMenu.OnMenuItemClickListener initMenu(int index) {
...
-                                else {
-                                    db.addToFriendList(ids[index]);
-                                    Toast.makeText(this, "friend added", Toast.LENGTH_SHORT).show();
-                                }
+                                db.addToFriendList(ids[index]);
+                                Toast.makeText(this,  "user added to your friend list", Toast.LENGTH_SHORT).show();
                             }
-                        });
+                        }
+                    });
```
Comment (Aco-Hub) [inline app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:189]: Why do you use a lambda? It seems like a lot of code that could be simpler

**Review by Aco-Hub — CHANGES_REQUESTED** [review]
Need some modifications/comments

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java [hunk 3]
```diff
@@ -154,29 +172,29 @@ private PopupMenu.OnMenuItemClickListener initMenu(int index) {
                     startActivity(intent);
                     return true;
                 case R.id.addFriendLeaderBoard:
-                        db.readField(mAuth.getCurrentUser().getUid(), "friends",task -> {
-                            if (!task.isSuccessful()) {
```
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:158]: use constant for the string friends

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java [hunk 4]
```diff
@@ -154,29 +172,29 @@ private PopupMenu.OnMenuItemClickListener initMenu(int index) {
...
-                                    Toast.makeText(this, "friend already added", Toast.LENGTH_SHORT).show();
-                                }
-                                else {
-                                    db.addToFriendList(ids[index]);
-                                    Toast.makeText(this, "friend added", Toast.LENGTH_SHORT).show();
-                                }
+                                db.addToFriendList(ids[index]);
+                                Toast.makeText(this,  "user added to your friend list", Toast.LENGTH_SHORT).show();
                             }
-                        });
```
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:171]: Use constant string

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 1 above)
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:45]: I think that STRING_MAX_LEN is unused

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardFragment.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
+        }
```
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardFragment.java:55]: what is this arguments that you get?? Is it usefull??

**Review by Antonio06188 — CHANGES_REQUESTED** [review]
Good code but you use the same string many time, you can use a constant for that

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 1 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:44]: yes, good observation thank you!

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 2 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:198]: This is the way firebase works, unfortunately nobody has found a way to do it differently.

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java [hunk 6]
```diff
@@ -228,4 +245,40 @@ public void onCancelled(@NonNull DatabaseError error) {
         });
     }
 
+    private void initTimer() {
```
[the remark this answers is not in the document]
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:271]: I added them, thank you

File: app/src/main/java/ch/epfl/sdp/healthplay/MonthlyLeaderBoardActivity.java [hunk 7]
```diff
@@ -0,0 +1,206 @@
+package ch.epfl.sdp.healthplay;
```
[the remark this answers is not in the document]
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/MonthlyLeaderBoardActivity.java:1]: done

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 3 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:158]: Aco has already commented on this, try looking at other people's comments before making yours!

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 4 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:171]: did it

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 1 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:44]: Aco has already commented on this, try looking at other people's comments before making yours!

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java (hunk 3 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardActivity.java:158]: done

File: app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardFragment.java (hunk 5 above)
Comment (emonnin-epfl) [reply app/src/main/java/ch/epfl/sdp/healthplay/LeaderBoardFragment.java:55]: the two leaderboards are still activities I did not have time to convert them into fragments

**Review by Aco-Hub — APPROVED** [review]

**Pull request merged**